src/pages/delivery-orders/Forms/DeliveryOrdersForm.js

```diff
@@ -0,0 +1,762 @@
+      printStatus: null,
+      invoiceStatus: null,
+      dtName: '',
+      plantName: '',
+      vehicleName: '',
+      driverName: '',
+      exWorks: false,
+      addressId: null,
+      szName: '',
+      szId: null,
+      spName: '',
+      spId: null,
+      volume: null,
+      qty: null,
+      siteId: null,
+      address: '',
+      orders: ordersInitialValues
+    },
+    maxAccess,
+    enableReinitialize: false,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      plantId: yup.number().required(),
+      siteId: yup.string().required(),
+      orders: yup.array().of(
+        yup.object({
+          mwRef: yup.string().required(),
+          siteName: yup.string().required(),
+        })
+      )
+    }),
+    onSubmit: async obj => {
+      const copy = { ...obj }
+      delete copy.orders
+
+      copy.date = formatDateToApi(copy.date)
+
+      const orders = formik.values.orders.map((order, index) => ({
+        ...order,
+        doId: recordId || 0,
+        doSeqNo: index + 1,
+        id: index + 1,
+        reference: order.mwRef 
+      }))
+
+      const data = {
+        header: copy,
+        items: orders
+      }
+
+      const res = await postRequest({
+        extension: DeliveryRepository.DeliveriesOrders.set2,
+        record: JSON.stringify(data)
+      })
+
+      formik.setFieldValue('recordId', res.recordId)
+
+      await refetchForm(res.recordId)
+      !formik.values.recordId ? toast.success(platformLabels.Added) : toast.success(platformLabels.Edited)
+
+      invalidate()
+    }
+  })
+
+  const totalQty = formik.values.orders.reduce((qtySum, row) => {
+    const qtyValue = parseFloat(row.qty) || 0
+
+    return qtySum + qtyValue
+  }, 0)
+
+  const isPosted = formik.values.status === 3
+  const isCancelled = formik.values.status == -1
+  const editMode = !!formik.values.recordId
+
+  async function refetchForm(recordId) {
+    const res = await getDeliveryOrder(recordId)
+
+    res.record.date = formatDateFromApi(res.record.date)
+    const doItems = await getOrders(res?.record)
+    await fillForm(res, doItems)
+  }
+
+  async function getDeliveryOrder(recordId) {
+    return await getRequest({
+      extension: DeliveryRepository.DeliveriesOrders.get,
+      parameters: `_doId=${recordId}&_recordId=${recordId}`
+    })
+  }
+
+  async function getAddress(addressId) {
+    if (!addressId) return null
+
+    const res = await getRequest({
+      extension: SystemRepository.FormattedAddress.get,
+      parameters: `_addressId=${addressId}`
+    })
+
+    return res?.record?.formattedAddress.replace(/(\r\n|\r|\n)+/g, '\r\n')
+  }
+
+  const getOrders = async data => {
+    return await getRequest({
+      extension: DeliveryRepository.OrderItem.qry,
+      parameters: `_tripId=${data.recordId}&_doId=${data.recordId}`
+    })
+  }
+
+  async function fillForm(doHeader, doItems) {
+    const address = await getAddress(doHeader?.record?.addressId)
+
+    let ordersList = []
```

Comment:
If no doItems are present, initialize this list with an obj contains id: 1 to avoid errors